Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -276,7 +285,14 @@ open class ComposerVC: _ViewController,
 
     override open func setUp() {
         super.setUp()
-
+        
+        if let cooldownDuration = channelController?.channel?.cooldownDuration,
+           cooldownDuration > 0 {
+            content.countdownDuration = cooldownDuration
+        }
```

Comment:
Since the countdown comes from `ChannelController.config` it is probably not a good use case to add a new `content.countdownDuration` property. **Unless**, the `content.countdownDuration` is used to track the current duration. So whenever the timer updates, it would update the `content.countdownTime`. If the logic of the timer lives insde the `CountdownButton`, we could have something like this: 

```
countdownButton.onCountdownChange = { countdown in
   self.content.countdownDuration = countdown
}
```

But since this introduces a new composer state, it actually makes sense for having a new state change function in `ComposerVC.Content.slowMode(countdown:)`

And so, whenever the countdown changes, it would call `content.slowMode(countdown:)`

Reply:
Done